### PR TITLE
Fix glusterfs'  backup-volfile-servers  option

### PR DESCRIPTION
**Pull request by nagexiucai**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

in pkg/volume/glusterfs/glusterfs.go:

  remove the selected ip from backup servers' ip set.

**Which issue(s) this PR fixes**:

"[glusterfsd.c:825:gf_remember_backup_volfile_server] 0-glusterfs: failed to set volfile server: File exists" when the selected ip is also in backups.

**Special notes for your reviewer**:

test with k8s v1.10.0 and glusterfs v6.1 on centos 7.2.1511(kernel 3.15.6 smp x86_64 gnu/linux).

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by nagexiucai** [comment]
Environment:

```
centos 7.2.1511 kernel 3.15.6 smp x86_64 gnu/linux
glusterfs: v6.1
kubernetes: v1.10.0
```

Description:

`mount -t glusterfs -o log-level=ERROR,log-file=/home/glusterfs.log,backup-volfile-servers=192.168.13.226:192.168.13.232:192.168.13.234 192.168.13.226:test /home/gfstest` will raises "[glusterfsd.c:825:gf_remember_backup_volfile_server] 0-glusterfs: failed to set volfile server: File exists".

`mount -t glusterfs -o log-level=ERROR,log-file=/home/glusterfs.log,backup-volfile-servers=192.168.13.232:192.168.13.234 192.168.13.226:test /home/gfstest` is ok.

And BTW:
After this PR, i had signed the CLA using my GitHub's account.
This is my first PR to K8S, if there may be any illegal behavior, please let me know and lend me a hand generously.

**Comment by jsafrane** [comment]
@nagexiucai, please don't use merge commits, rebase your branch on top of current master instead. As result, there should be just one commit in this PR.

/ok-to-test
/assign @humblec 
for gluster expertise

File: pkg/volume/glusterfs/glusterfs.go [hunk 1]
```diff
@@ -383,19 +383,24 @@ func (b *glusterfsMounter) setUpAtInternal(dir string) error {
 
 	}
 
-	//Add backup-volfile-servers and auto_unmount options.
-	options = append(options, "backup-volfile-servers="+dstrings.Join(addrlist[:], ":"))
-	options = append(options, "auto_unmount")
+	if (len(addrlist) > 0) && (addrlist[0] != "") {
+		ip := addrlist[rand.Intn(len(addrlist))]
```
Comment (humblec) [inline pkg/volume/glusterfs/glusterfs.go]: This part is already in. Please rebase to master and update this PR.

**Comment by humblec** [comment]
>"[glusterfsd.c:825:gf_remember_backup_volfile_server] 0-glusterfs: failed to set volfile server: File exists" when the selected ip is also in backups.
>
@nagexiucai its just a warning , isnt it ? afiak, no functional issues with this.

File: pkg/volume/glusterfs/glusterfs.go (hunk 1 above)
Comment (nagexiucai) [reply pkg/volume/glusterfs/glusterfs.go]: OK, i will update it.

**Comment by nagexiucai** [comment]
@humblec right, it is a warning. the 'E' puzzles me.

[](https://github.com/nagexiucai/kubernetesex/blob/master/contributions/illustrations/fix-glusterfs--backup-volfile-servers--option.gif)

**Comment by nagexiucai** [comment]
@jsafrane @humblec sorry, i had deleted the branch and redo a same pr.

so, it is not a bug, what kind it can be?

**Comment by nagexiucai** [comment]
> @nagexiucai, please don't use merge commits, rebase your branch on top of current master instead. As result, there should be just one commit in this PR.

i had done:

```
git fetch upstream
git checkout master
git merge upstream/master
git checkout fix-glusterfs--backup-volfile-servers--option
git rebase master
git push origin --delete fix-glusterfs--backup-volfile-servers--option
git push origin fix-glusterfs--backup-volfile-servers--option:fix-glusterfs--backup-volfile-servers--option
```

but the pr #77769 is still with four commits.

how to make only one commit in this pr?

File: pkg/volume/glusterfs/glusterfs.go (hunk 1 above)
Comment (nagexiucai) [reply pkg/volume/glusterfs/glusterfs.go]: Move lines of `options=...` and comments below `if (len(addrlist) > 0) && (addrlist[0] != "")` for two reasons: `addr.Delete(ip)` uses `ip` and "options" is only used in this condition.
I think "This part" here "minus/plus" is just git's difference.
Maybe i do not get your idea.

**Pull request closed without merging**